test/e2e/performance_basic_test.go

```diff
@@ -0,0 +1,204 @@
+//go:build performance_tests
+
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/utils/kubernetes/generators"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/test"
+	"github.com/kong/kubernetes-ingress-controller/v3/test/internal/helpers"
+)
+
+var (
+	defaultResNum = 10000
+	resNumStr     = os.Getenv("PERF_RES_NUMBER")
```

Comment:
This seems to be rather important but has a rather cryptic name and no comment. Can we add a godoc for this?

Reply:
thanks, I will add it